saw-core/src/SAWCore/SCTypeCheck.hs

```diff
@@ -90,7 +92,7 @@ type TCState = Map TermIndex Term
 --
 -- * Can throw 'TCError's
 type TCM =
-  ReaderT (SharedContext, Maybe ModuleName, [(LocalName, Term)])
+  ReaderT (SharedContext, Maybe ModuleName, [(LocalName, Term)], Map LocalName (ExtCns Term))
```

Comment:
There are enough elements in this tuple that I think it would be worth factoring this out into a standalone data type.

Reply:
Fixed in 965fe21f2965b3178a470b4ecec8b28128810ca9.